Purchases/Purchasing/TrialOrIntroPriceEligibilityChecker.swift

```diff
@@ -92,6 +93,23 @@ class TrialOrIntroPriceEligibilityChecker {
         return introDict
     }
 
+    @available(iOS 11.2, macOS 10.13.2, macCatalyst 13.0, tvOS 11.2, watchOS 6.2, *)
+    func productsWithIntroOffers(productIdentifiers: [String], completion: @escaping ReceiveIntroEligibilityBlock) {
+        self.productsManager.products(withIdentifiers: Set(productIdentifiers)) { products in
+            let eligibility: [(String, IntroEligibility)] = Array(products).compactMap {
+                guard $0.introductoryPrice != nil else {
+                    return nil
+                }
+                return ($0.productIdentifier, IntroEligibility(eligibilityStatus: .eligible))
+            }
+            var productIdsToIntroEligibleStatus: [String: IntroEligibility] = [:]
+            eligibility.forEach { (productId, eligibility) in
+                productIdsToIntroEligibleStatus[productId] = eligibility
+            }
```

Comment:
I think this could be simplified using [this init](https://developer.apple.com/documentation/swift/dictionary/3127165-init)
```swift
let productIdsToIntroEligibleStatus = Dictionary(uniqueKeysWithValues: eligibility)
```

Reply:
I totally forgot this was a thing!

Reply:
✅